### PR TITLE
Max multisampling support and refactoring

### PR DESCRIPTION
**Pull request by carlOSx64**
Unfortunately, this is not the awaited `sunshine.jpg` commit.

I wanted to clean things up a bit before works on the NIfTI format begins.

- Added `file.rs` as a core module and removed `save.rs`. This creates a separation between file IO and GPU operations.
- Improved `expect(..)` messages to better follow the recommended message style (see https://doc.rust-lang.org/stable/std/error/index.html#common-message-styles) (PS: Error handling in some places could be improved in the future, we can talk about this later)
- `graphics::Context` is now private, `run_full_pipeline()` serves as the only entry point into the `graphics` module.
- Added support for higher multisample count when available on the device. 4x is always supported but some devices allows for more (mine has 8x). We could run our own AA solution down the line if necessary so quality stays consistent across devices. But what is provided here get the job done at a low cost. This will only affect streamlines and glyphs rendering.

### CODE REVIEW

File: src/graphics/pipeline.rs [hunk 1]
```diff
@@ -84,9 +85,9 @@ fn layout(name: &str, device: &Device) -> wgpu::PipelineLayout {
     })
 }
 
-fn multisample(alpha_to_coverage_enabled: bool) -> wgpu::MultisampleState {
+fn multisample(count: u32, alpha_to_coverage_enabled: bool) -> wgpu::MultisampleState {
     wgpu::MultisampleState {
-        count: MULTISAMPLE_COUNT,
+        count,
         mask: !0,
```
Comment (nilgoyette) [inline src/graphics/pipeline.rs:91]: Hahaha, what? Not zero? Is that an integer filled with ones?

Sorry, I know it's not new, but I managed to not see it in the previous reviews!

**Review by nilgoyette — APPROVED** [review]

File: src/graphics/pipeline.rs (hunk 1 above)
Comment (carlOSx64) [reply src/graphics/pipeline.rs:91]: I'm also confused by this, here's the passage from the doc :

> Bitmask that restricts the samples of a pixel modified by this pipeline. All samples can be enabled using the value !0

Comment (carlOSx64) [reply src/graphics/pipeline.rs:91]: I looked it up. This is indeed a bitmask with 64 ones

1111111111111111111111111111111111111111111111111111111111111111

Comment (nilgoyette) [reply src/graphics/pipeline.rs:91]: So it's an integer filled with ones, identical to `int i = -1;`. Thanks.

**Pull request merged**